aws/resource_aws_appautoscaling_policy_test.go

```diff
@@ -310,6 +311,44 @@ func TestAccAWSAppautoScalingPolicy_ResourceId_ForceNew(t *testing.T) {
 	})
 }
 
+func TestValidateAppautoscalingPolicyImportInput(t *testing.T) {
+	testCases := []struct {
+		input                     string
+		expectedServiceNamespace  string
+		expectedResourceId        string
+		expectedScalableDimension string
+		expectedPolicyName        string
+	}{
+		{
+			"rds/cluster:id/rds:cluster:ReadReplicaCount/cpu-auto-scaling",
+			"rds", "cluster:id", "rds:cluster:ReadReplicaCount", "cpu-auto-scaling"},
+		{
+			"ecs/service/clusterName/serviceName/ecs:service:DesiredCount/scale-down",
+			"ecs", "service/clusterName/serviceName", "ecs:service:DesiredCount", "scale-down"},
+		{
+			"dynamodb/table/tableName/dynamodb:table:ReadCapacityUnits/DynamoDBReadCapacityUtilization:table/tableName",
+			"dynamodb", "table/tableName", "dynamodb:table:ReadCapacityUnits", "DynamoDBReadCapacityUtilization:table/tableName"},
+	}
+
+	for _, test := range testCases {
+		idParts := strings.Split(test.input, "/")
+		serviceNamespace, resourceId, scalableDimension, policyName := validateAppautoscalingPolicyImportInput(idParts)
+
+		if serviceNamespace != test.expectedServiceNamespace {
+			t.Errorf("serviceNamespace interpolation error -> %s != %s", test.expectedServiceNamespace, serviceNamespace)
+		}
+		if resourceId != test.expectedResourceId {
+			t.Errorf("resourceId interpolation error -> %s != %s", test.expectedResourceId, resourceId)
+		}
+		if scalableDimension != test.expectedScalableDimension {
+			t.Errorf("scalableDimension interpolation error -> %s != %s", test.expectedScalableDimension, scalableDimension)
+		}
+		if policyName != test.expectedPolicyName {
+			t.Errorf("policyName interpolation error -> %s != %s", test.expectedPolicyName, policyName)
+		}
+	}
+}
+
```

Comment:
Given your implementation details I would expect there being a complimentary change to `testAccCheckAWSAppautoscalingPolicyExists` as well. I will try to reproduce the exact issue to better understand the behaviour. So please let me know if this is an incorrect assumption.

Reply:
No, I think `testAccCheckAWSAppautoscalingPolicyExists` is out of scope as its validating the state file. Creation works fine, and the resources are correctly mapped. The problem occurs only while mapping the input in `resourceAwsAppautoscalingPolicyImport` (With my change, I was able to import several DynamoDB policies and continue using them to make changes with no issues)